src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -117,24 +117,28 @@ std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> ObjectBufferPool::Cr
         plasma::flatbuf::ObjectSource::ReceivedFromRemoteRaylet);
     lock.lock();
```

Comment:
The race condition is introduced by releasing the lock above. We need to double-check the condition upon re-entering the block.